File: pset1/mario/more/mario.c
/*
* Jeffrey Chiu
* CS-50-Problem Set 1 mario.c
* Generates a full-pyramid like that in Mario using nested for loops
*/
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height = -1;
    //make sure height constraints are met
    while (height < 0 || height > 23)
    {
        printf("Height: ");
        height = get_int();
    }
    //loop though each layer one by one
    for (int i = 0; i < height; i++)
    {
        //prints the initial empty spaces
        for (int j = height - 1 - i; j > 0; j--)
        {
            printf(" ");
        }
        //print the hashmarks on the left side
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }
        //print the spaces in between each side
        printf("  ");
        //print the hashmarks on the right side
        for (int m = 0; m <= i; m++)
        {
            printf("#");
        }
        //on to the next line
        printf("\n");
    }
}

File: pset6/mario/mario.py
"""
 * CS50 Pset 6
 * mario.py
 *
 * Author: Jeffrey Chiu
 * mario/more from pset1 in python
 *
"""

from cs50 import get_int

# Function to ensure entered height is within [0, 23]
while(True):
    height = get_int("Height: ")
    if height > 0 and height <= 23:
        break

for i in range(height):
    # print proper starting spaces
    for j in range(height - i - 1, 0, -1):
        print(" ", end="")

    # print proper number of hashtags on left side
    for l in range(i + 1):
        print("#", end="")

    # print spaces in between
    print("  ", end="")

    # print hashtags on right side
    for m in range(i + 1):
        print("#", end="")

    # new line!
    print()
